app/models/post.rb

```diff
@@ -1,2 +1,17 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  # A method that updates the posts counter for a user.
+
+  def updates_post_counter_user
+    author.increment!(:post_counter)
+  end
```

Comment:
- [ ] [optional] it would have been nice if you used [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, this will allow you to make your  method private